[nlopt] Adjust the code for vendor_cxx compatibility

When weaving the inline namespace into this file, we need to ensure that
it doesn't cross an `#if 0` block.

Reasoning for not upstreaming this patch: Drake-specific vendoring.

--- src/algs/stogo/local.cc
+++ src/algs/stogo/local.cc
@@ -11,11 +11,12 @@
 #include "local.h"
 #include "tools.h"
 
+typedef struct {} force_the_drake_inline_namespace_to_appear_here;
+
 ////////////////////////////////////////////////////////////////////////
 // SGJ, 2007: allow local to use local optimizers in NLopt, to compare
 // to the BFGS code below
 #if 0
-#include "nlopt.h"
 
 typedef struct {
   Global *glob;
